osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,184 @@
+    {GENERIC_READ, "Generic Read"},
+    {GENERIC_WRITE, "Generic Write"},
+    {GENERIC_EXECUTE, "Generic Execute"},
+    {GENERIC_ALL, "Generic All"}};
+
+// map to get inheritance string
+static std::unordered_map<unsigned long, std::string> inheritanceToStr = {
+    {CONTAINER_INHERIT_ACE, "Container Inherit Ace"},
+    {INHERIT_NO_PROPAGATE, "Inherit No Propagate"},
+    {INHERIT_ONLY, "Inherit Only"},
+    {OBJECT_INHERIT_ACE, "Object Inherit Ace"},
+    {SUB_CONTAINERS_AND_OBJECTS_INHERIT, "Sub containers and Objects Inherit"},
+    {NO_INHERITANCE, "No Inheritence"}};
+
+// helper function to build access string from permission bit mask
+std::string accessPermsToStr(const unsigned long pmask) {
+  std::vector<std::string> permList;
+
+  for (auto const& perm : permVals) {
+    if ((pmask & perm.first) != 0) {
+      permList.push_back(perm.second);
+    }
+  }
+
+  return alg::join(permList, ",");
+}
+
+// helper function to get account/group name from trustee
+std::string trusteeToStr(const TRUSTEE& trustee) {
+  // Max username length for Windows
+  int r;
+  const unsigned long kMaxBuffSize = 256;
+  char name[kMaxBuffSize];
+  char domain[kMaxBuffSize];
+  unsigned long size = kMaxBuffSize;
+  SID_NAME_USE accountType;
+
+  switch (trustee.TrusteeForm) {
+  case TRUSTEE_IS_SID: {
+    // get the name from the SID
+    PSID psid = trustee.ptstrName;
+    r = LookupAccountSid(
+        nullptr, psid, name, &size, domain, &size, &accountType);
+    if (!r) {
+      VLOG(1) << "LookupAccountSid error: " << GetLastError();
+    } else {
+      return name;
+    }
+  }
+  case TRUSTEE_IS_NAME:
+    // get the name from ptstrName
+    return trustee.ptstrName;
+  case TRUSTEE_BAD_FORM:
+    // Indicates a trustee form that is not valid.
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/aa379638(v=vs.85).aspx
+    return "Invalid";
+  case TRUSTEE_IS_OBJECTS_AND_SID: {
+    // ptstrName member is a pointer to an OBJECTS_AND_SID struct
+    PSID psid = reinterpret_cast<POBJECTS_AND_SID>(trustee.ptstrName)->pSid;
+    r = LookupAccountSid(
+        nullptr, psid, name, &size, domain, &size, &accountType);
+    if (!r) {
+      VLOG(1) << "LookupAccountSid error: " << GetLastError();
+    } else {
+      return name;
+    }
+  }
+  case TRUSTEE_IS_OBJECTS_AND_NAME:
+    // ptstrName member is a pointer to an OBJECTS_AND_NAME struct
+    return reinterpret_cast<OBJECTS_AND_NAME_*>(trustee.ptstrName)->ptstrName;
+  default:
+    return "";
+  }
+}
+
+QueryData genNtfsAclPerms(QueryContext& context) {
+  QueryData results;
+  unsigned long result = 0;
+  PACL dacl = nullptr;
+  PEXPLICIT_ACCESS aceList = nullptr;
+  unsigned long aceCount = 0;
+
+  auto paths = context.constraints["path"].getAll(EQUALS);
+  for (const auto& pathString : paths) {
+    if (!fs::exists(pathString)) {
+      continue;
+    }
+    // Get a pointer to the existing DACL.
+    result = GetNamedSecurityInfo(pathString.c_str(),
```

Comment:
You don't need to define result in advance at the beginning of the function. You can use just 
```C++
auto result = GetNamedSecurityInfo(...);
```